### PR TITLE
Update Docker build

### PR DESCRIPTION
**Pull request by SuperQ**
* Update Dockerfile for external builds.
* Remove Docker from dependabot, not necessary with external build.
* Rework CI to match Prometheus Community setup.
* Update gitignore for new build.

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -3,9 +3,16 @@ on: push
 name: Build and test
 jobs:
   buildDockerImage:
-    name: Build docker image
+    name: Publish master branch artifacts
     runs-on: ubuntu-latest
+    needs: [test_ui, test_go, test_go_more, test_go_oldest, test_windows, golangci, codeql, build_all]
```
Comment (SuperQ) [inline .github/workflows/push.yml:8]: Oh, this is copy-pasta from prometheus. I'll update it.

File: .yamllint [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (cristiangreco) [inline .yamllint:1]: Do we want to have a CI check for linting yaml?

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
+.build
 yet-another-cloudwatch-exporter
 !charts/yet-another-cloudwatch-exporter
 vendor
 dist
 /yace
-.idea
+*.tar.gz
```
Comment (cristiangreco) [inline .gitignore:7]: We could still continue to gitignore the `.idea` folder I guess?

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -3,9 +3,16 @@ on: push
...
-    - name: Build docker image
-      run: docker build -t yace --build-arg VERSION=${{github.event.release.tag_name}} .
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: prometheus/promci@52c7012f5f0070d7281b8db4a119e21341d43c91 # v0.4.5
+      - uses: ./.github/promci/actions/publish_master
+        with:
+          docker_hub_login: ${{ secrets.docker_hub_login }}
+          docker_hub_password: ${{ secrets.docker_hub_password }}
+          quay_io_login: ${{ secrets.quay_io_login }}
+          quay_io_password: ${{ secrets.quay_io_password }}
```
Comment (cristiangreco) [inline .github/workflows/push.yml:18]: Where are these secrets coming from? Are those already setup in this repo?

**Review by cristiangreco — COMMENTED** [review]
Thanks @SuperQ! Looks good, left some comment just to better understand some details that are new to me as this repo is starting to use CI toolings shared with prometheus-community.

File: .github/workflows/push.yml (hunk 4 above)
Comment (SuperQ) [reply .github/workflows/push.yml:18]: They come from the prometheus community GitHub org vault.

File: .yamllint (hunk 2 above)
Comment (SuperQ) [reply .yamllint:1]: It does via the `make lint`, which is part of the `make all` target.

File: .gitignore (hunk 3 above)
Comment (SuperQ) [reply .gitignore:7]: We have a community policy to not include these editor patterns in our configuration. There are just too many variations to support.

This is also better handled by the developer's git configuration. We have a "canned response" for theses requests.

```
Please don't include your local editor ignores in the repo. Use [`git config --global core.excludesFile`](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/).
```

**Review by cristiangreco — APPROVED** [review]
Thanks! 🚀 🚀 🚀

**Pull request merged**